Change default GTP clock source for the APF51 to 32k one. Indeed current U-Boot
setup 32k as main clock for GPT1 and Linux should take that into account.

Patch has to be reworked to dynamically do that switch.

Signed-off-by: Nicolas Colombain <nicolas.colombain@armadeus.com>

Index: linux-2.6.38-alpha/arch/arm/mach-mx5/clock-mx51-mx53.c
===================================================================
--- linux-2.6.38-alpha.orig/arch/arm/mach-mx5/clock-mx51-mx53.c	2010-12-29 10:34:04.000000000 +0100
+++ linux-2.6.38-alpha/arch/arm/mach-mx5/clock-mx51-mx53.c	2010-12-29 10:35:10.000000000 +0100
@@ -1322,7 +1322,7 @@
 	clk_set_rate(&esdhc2_clk, 166250000);
 
 	/* System timer */
-	mxc_timer_init(&gpt_clk, MX51_IO_ADDRESS(MX51_GPT1_BASE_ADDR),
+	mxc_timer_init(&gpt_32k_clk, MX51_IO_ADDRESS(MX51_GPT1_BASE_ADDR),
 		MX51_MXC_INT_GPT);
 	return 0;
 }
Index: linux-2.6.38-alpha/arch/arm/plat-mxc/time.c
===================================================================
--- linux-2.6.38-alpha.orig/arch/arm/plat-mxc/time.c	2010-12-29 10:33:53.000000000 +0100
+++ linux-2.6.38-alpha/arch/arm/plat-mxc/time.c	2010-12-29 10:35:10.000000000 +0100
@@ -56,6 +56,7 @@
 /* MX31, MX35, MX25, MXC91231, MX5 */
 #define V2_TCTL_WAITEN		(1 << 3) /* Wait enable mode */
 #define V2_TCTL_CLK_IPG		(1 << 6)
+#define V2_TCTL_CLK_IPG_32k	(1 << 8)
 #define V2_TCTL_FRR		(1 << 9)
 #define V2_IR			0x0c
 #define V2_TSTAT		0x08
@@ -120,7 +121,7 @@
 	.rating		= 200,
 	.read		= mx1_2_get_cycles,
 	.mask		= CLOCKSOURCE_MASK(32),
-	.shift 		= 20,
+	.shift 		= 10, /* kHz */
 	.flags		= CLOCK_SOURCE_IS_CONTINUOUS,
 };
 
@@ -135,6 +136,8 @@
 					clocksource_mxc.shift);
 	clocksource_register(&clocksource_mxc);
 
+	printk("___%s %dkHz mult=%d\n", __func__, c, clocksource_mxc.mult);
+
 	return 0;
 }
 
@@ -285,6 +288,7 @@
 	clockevent_mxc.min_delta_ns =
 			clockevent_delta2ns(0xff, &clockevent_mxc);
 
+	printk("___ %s %dkHz mult=%d\n", __func__, c, clockevent_mxc.mult);
 	clockevent_mxc.cpumask = cpumask_of(0);
 
 	clockevents_register_device(&clockevent_mxc);
@@ -308,7 +312,8 @@
 	__raw_writel(0, timer_base + MXC_TPRER); /* see datasheet note */
 
 	if (timer_is_v2())
-		tctl_val = V2_TCTL_CLK_IPG | V2_TCTL_FRR | V2_TCTL_WAITEN | MXC_TCTL_TEN;
+		tctl_val = V2_TCTL_CLK_IPG_32k | V2_TCTL_FRR | V2_TCTL_WAITEN | MXC_TCTL_TEN;
+//		tctl_val = V2_TCTL_CLK_IPG | V2_TCTL_FRR | V2_TCTL_WAITEN | MXC_TCTL_TEN;
 	else
 		tctl_val = MX1_2_TCTL_FRR | MX1_2_TCTL_CLK_PCLK1 | MXC_TCTL_TEN;
 
